native-plasma/app/src/main/jni/plasma.c

```diff
@@ -427,15 +427,26 @@ static int32_t engine_handle_input(struct android_app* app, AInputEvent* event)
 }
 
 static void engine_handle_cmd(struct android_app* app, int32_t cmd) {
+    static int32_t format = WINDOW_FORMAT_RGB_565;
     struct engine* engine = (struct engine*)app->userData;
     switch (cmd) {
         case APP_CMD_INIT_WINDOW:
             if (engine->app->window != NULL) {
+                // fill_plasma() assumes 565 format, get it here
+                format = ANativeWindow_getFormat(app->window);
+                ANativeWindow_setBuffersGeometry(app->window,
+                              ANativeWindow_getWidth(app->window),
+                              ANativeWindow_getHeight(app->window),
+                              WINDOW_FORMAT_RGB_565);
                 engine_draw_frame(engine);
             }
             break;
         case APP_CMD_TERM_WINDOW:
             engine_term_display(engine);
+            ANativeWindow_setBuffersGeometry(app->window,
+                          ANativeWindow_getWidth(app->window),
+                          ANativeWindow_getHeight(app->window),
+                          format);
```

Comment:
yes, thought about that:  adding a few lines there to make sure we are a good citizen on the system and making no assumptions on what system is doing:
  system gives us a window, with some format
  system get back from us the same window with the same format
in case it is crashed somewhere and system wants to output some error message on that window in its final stage after calling _TERM_WINDOW, he could still do it. 

for this app, there is no effect whether we restore or not.